### PR TITLE
test: add unit tests for pkg/tls

### PR DESCRIPTION
**Pull request by magic-peach**
## Purpose of this PR

pkg/tls parses the --tls-min-version and --tls-cipher-suites flags used by both the webhook and controller manager startup paths (cmd/operator/webhook/start.go and cmd/operator/controller/start.go), but had no test coverage at all.

**Proposed changes:**

- Add unit tests for ParseTLSVersion covering supported and unsupported version strings
- Add unit tests for ParseCipherSuites covering nil/empty input, a valid cipher suite name, and an unknown name
- Add unit tests for SetupTLS covering invalid input rejection and that the returned options correctly configure a tls.Config

## Change Category

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that could affect existing functionality)
- [ ] Documentation update

### Rationale

Test-only change, no behavior modified.

## Checklist

- [x] I have conducted a self-review of my own code.
- [ ] I have updated documentation accordingly.
- [x] I have added tests that prove my changes are effective or that my feature works.
- [x] Existing unit tests pass locally with my changes.

### Additional Notes

No existing issue tracks this; found while reading the TLS setup code.

### CODE REVIEW

**Review by tariq-hasan — CHANGES_REQUESTED** [review]
Thanks for the PR @magic-peach! Please rewrite the tests using Ginkgo/Gomega as that would be the default going forward.

File: pkg/tls/tls_test.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+	It("leaves CipherSuites unset when none are given", func() {
+		opts, err := tls.SetupTLS("VersionTLS12", nil)
+		Expect(err).NotTo(HaveOccurred())
+
+		cfg := &cryptotls.Config{}
+		for _, opt := range opts {
+			opt(cfg)
+		}
+
+		Expect(cfg.CipherSuites).To(BeNil())
```
Comment (RobuRishabh) [inline pkg/tls/tls_test.go:119]: Nil-cipher path under-asserts, Also pin MinVersion and NextProtos.

File: pkg/tls/tls_test.go [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+		Expect(opts).NotTo(BeEmpty())
+
+		cfg := &cryptotls.Config{}
+		for _, opt := range opts {
+			opt(cfg)
+		}
+
+		Expect(cfg.MinVersion).To(Equal(uint16(cryptotls.VersionTLS13)))
+		Expect(cfg.CipherSuites).To(Equal([]uint16{suites[0].ID}))
+		Expect(cfg.NextProtos).To(Equal([]string{"h2", "http/1.1"}))
```
Comment (RobuRishabh) [inline pkg/tls/tls_test.go:107]: Happy path should use TLS 1.2. Matches flag defaults; cipher suites actually apply there.

**Comment by RobuRishabh** [comment]
Hi @magic-peach can you address the above comments. Thanks
